api/RS.API/RS.Service/Logic/CandidateManagerService.cs

```diff
@@ -315,31 +343,50 @@ public IResult UpdateCandidate(CandidateViewModel candidate)
                 {
-                    candidateModel.OrganizationId = organizationModel.OrganizationId;
+                    var organizationModel = _candidateRepository.GetOrganization(candidateViewModel.OrganizationName);
+                    if (organizationModel != null)
+                    {
+                        candidateModel.OrganizationId = organizationModel.OrganizationId;
+                    }
+                    else
+                    {
+                        organization = new Organizations
+                        {
+                            Name = candidateViewModel.OrganizationName
+                        };
+                        organization.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+                        candidateModel.Organisation = organization;
+                    }
                 }
+                #endregion
 
-                var organization = new Organizations();
-                organization.Name = candidate.Organization;
-                organization.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+                #region OpeningCandidate Update
+                var openingCandidateModel = _candidateRepository.GetOpeningCandidate(candidateViewModel.CandidateId);
+                if ((openingCandidateModel != null) && (openingCandidateModel.Opening.OpeningId != candidateViewModel.OpeningId))
+                {
+                    openingCandidateModel.OpeningId = candidateViewModel.OpeningId;
+                    openingCandidateModel.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+                } 
+                #endregion
 
-                var openingCandidate = _candidateRepository.GetOpeningCandidate(candidate.CandidateId);
-                var updatedOpeningCandidate = new OpeningCandidates();
-                if ((openingCandidate != null) && (openingCandidate.Opening.OpeningId != candidate.Opening))
+                #region Candidate Document Insert Update
+                var candidateDocumentModel = candidateModel.CandidateDocuments.FirstOrDefault();
+                if (candidateDocumentViewModel != null)
                 {
-                    openingCandidate.MapDeleteColumns((ClaimsIdentity)_principal.Identity);
-                    updatedOpeningCandidate.CandidateId = candidateModel.CandidateId;
-                    updatedOpeningCandidate.OpeningId = candidate.Opening;
-                    updatedOpeningCandidate.MapAuditColumns((ClaimsIdentity)_principal.Identity);
-                }
-                _candidateRepository.UpdateCandidate(candidateModel, updatedOpeningCandidate, organization);
-                result.Body = candidate.CandidateId;
+                    candidateDocumentModel.MapFromViewModel(candidateDocumentViewModel, (ClaimsIdentity)_principal.Identity);
+                } 
+                #endregion
+
+                _candidateRepository.UpdateCandidate(candidateModel, openingCandidateModel, candidateDocumentModel);
```

Comment:
attach candidatedocument model to candidate model as they have navigation

Reply:
Done